### PR TITLE
Cloud calls-to-action -> Enterprise

### PR DESCRIPTION
**Pull request by sqs**
This implements the proposed new [product naming and hierarchy](https://docs.google.com/document/d/1te8D2hsa0UPdEUFKda6W2jy5RsaQx5JgZgvb2oRcCRI/edit#). In this new naming scheme, "Sourcegraph Enterprise Cloud" is our preferred way for users to continue, but we'll just refer to "Sourcegraph Enterprise" in most cases (which technically also includes self-hosted).

Also changes links from https://signup.sourcegraph.com to https://about.sourcegraph.com.

## Test plan

Test with `SOURCEGRAPHDOTCOM_MODE=true` and see that CTAs are as described above.

## App preview:

- [Web](https://sg-web-sqs-enterprise-ctas.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| -0.02% (-0.60 kb) | -0.01% (-0.75 kb) | -0.00% (-0.15 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/sqs/enterprise-ctas/compare-report.html;tab=live_object) for a full comparison between the commits 0917373 and 1445abe or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: client/web/src/nav/GlobalNavbar.tsx [hunk 1]
```diff
@@ -340,14 +334,11 @@ export const GlobalNavbar: React.FunctionComponent<React.PropsWithChildren<Globa
...
-                                        size="sm"
-                                        onClick={() => eventLogger.log('ClickedOnTopNavTrialButton')}
-                                    >
-                                        Sign up
-                                    </ButtonLink>
+                                    {!isSourcegraphDotCom && (
+                                        <ButtonLink to="/sign-up" variant="primary" size="sm">
+                                            Sign up
+                                        </ButtonLink>
+                                    )}
```
Comment (philipp-spiess) [inline client/web/src/nav/GlobalNavbar.tsx:341]: Should this be behind a `context.allowSignup` check?

**Review by philipp-spiess — APPROVED** [review]

File: client/web/src/nav/GlobalNavbar.tsx (hunk 1 above)
Comment (sqs) [reply client/web/src/nav/GlobalNavbar.tsx:341]: Yes! It was not behind an `allowSignup` check before, which is strange. But I will add one.

File: client/branded/src/search-ui/input/SearchContextMenu.tsx [hunk 2]
```diff
@@ -256,14 +255,14 @@ export const SearchContextMenu: FC<SearchContextMenuProps> = props => {
                                 <Text className="mb-0">
                                     To search across your team's private repositories,{' '}
                                     <Link
-                                        to={buildCloudTrialURL(authenticatedUser, 'context')}
+                                        to="https://about.sourcegraph.com"
                                         onClick={() =>
-                                            telemetryService.log('ClickedOnCloudCTA', {
-                                                cloudCtaType: 'ContextDropDown',
+                                            telemetryService.log('ClickedOnEnterpriseCTA', {
```
Comment (limitedmage) [inline client/branded/src/search-ui/input/SearchContextMenu.tsx:260]: Note that renaming the events will make it harder to review historical data if that's important.

**Pull request merged**